zkevm-circuits/src/evm_circuit/witness.rs

```diff
@@ -24,19 +26,17 @@ pub struct Block<F> {
 
 #[derive(Debug, Default)]
 pub struct Transaction<F> {
-    // Context
     pub id: usize,
     pub nonce: u64,
     pub gas: u64,
-    pub gas_tip_cap: Word,
-    pub gas_fee_cap: Word,
+    pub gas_price: Word,
     pub caller_address: Address,
     pub callee_address: Address,
     pub is_create: bool,
     pub value: Word,
-    pub call_data_length: usize,
     pub call_data: Vec<u8>,
-
+    pub call_data_length: usize,
+    pub call_data_gas_cost: u64,
```

Comment:
what is the intention of call_data_gas_cost here?  I
 think the `gas` field counts it , or for call data out of gas purpose ?

Reply:
It makes `BeginTx` more complicated for EVM circuit to checks `call_data` gas cost, because we need to iterate over it byte to byte to check if it's zero or not. So currently we assume this information to be prepared in `tx_table`, and it could be calculated by verifier, or verified by some other constraint enforced on `tx_table`.

Reply:
Should we add this kind of work as a `bus-mapping` responsability?

Reply:
Yes eventually we should let `bus-mapping` to calculate it, but it would change a lot on `bus-mapping` to fully support generating the witness for `BeginTx`, which is already explored in further PR of `CALL` implementation. Is it acceptable to do this after that PR?

Reply:
Of course!! Otherways it would become massive! Makes complete sense!